angular/lib/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,539 @@
+import type {SliderHandleState, SliderState} from '@agnos-ui/angular-headless';
+import {callWidgetFactory, createSlider, patchSimpleChanges, toSlotContextWidget, UseDirective} from '@agnos-ui/angular-headless';
+import type {SliderActions} from '@agnos-ui/core';
+import {NgFor, NgIf} from '@angular/common';
+import type {AfterViewInit, OnChanges, OnDestroy, Signal, SimpleChanges} from '@angular/core';
+import {
+	ChangeDetectionStrategy,
+	Component,
+	DestroyRef,
+	ElementRef,
+	EventEmitter,
+	forwardRef,
+	inject,
+	Input,
+	NgZone,
+	Output,
+	ViewChild,
+	ViewEncapsulation,
+} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
+import {NG_VALUE_ACCESSOR} from '@angular/forms';
+import {take} from 'rxjs';
+
+/**
+ * A class that is responsible for:
+ * the click and keyboard even capture and propagation
+ * positioning of the slider handle
+ */
+@Component({
+	selector: '[auSliderHandle]',
+	exportAs: 'auSliderHandle',
+	standalone: true,
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	host: {
+		class: 'au-slider-handle',
+		'[class]': "state().vertical ? 'au-slider-handle-vertical' : 'au-slider-handle-horizontal'",
+		'[style.left.%]': 'state().positionLeft[state().handleId]',
+		'[style.top.%]': 'state().positionTop[state().handleId]',
+		'(keydown)': 'widget.actions.onKeyDown($event, state().handleId)',
+		'(mousedown)': 'widget.actions.onMouseDown($event)',
+	},
+	hostDirectives: [
+		{
+			directive: UseDirective,
+		},
+	],
+	template: ``,
+})
+export class SliderHandleComponent {
+	private _elementRef = inject(ElementRef);
+	private _zone = inject(NgZone);
+	// needed to avoid changedetection on the mouse move over the same coordinate (vertical or horizonal movement)
+	private _prevCoordinate = -1;
+
+	// TODO how to declare it in props?
+	// eslint-disable-next-line @agnos-ui/angular-check-props
+	@Input('auSliderActions') sliderActions: SliderActions;
+
+	/**
+	 * The id of the slider handle.
+	 */
+	@Input('auHandleId') handleId: number | undefined;
```

Comment:
Why have all these inputs? Everyhing comes from the state, no?

Reply:
if these inputs are not here, the build doesn't pass and I don't think that putting the ignore is the best idea

Reply:
I don't think the slider handle should be a widget.

Reply:
it is a widget because we need to pass the implementation on the onMouseDown and onKeydown per framework, so it is a prop, otherwise I don't see how we can override it